client/lib/account-settings-helper/index.jsx

```diff
@@ -0,0 +1,43 @@
+import languages from '@automattic/languages';
+import ReactDom from 'react-dom';
+import { Provider, useDispatch, useSelector } from 'react-redux';
+import QueryUserSettings from 'calypso/components/data/query-user-settings';
+import LanguagePicker from 'calypso/components/language-picker';
+import getUserSettings from 'calypso/state/selectors/get-user-settings';
+import { setUserSetting } from 'calypso/state/user-settings/actions';
+import { saveUnsavedUserSettings } from 'calypso/state/user-settings/thunks';
+
+import './style.scss';
+
+export function AccountSettingsHelper() {
+	const dispatch = useDispatch();
+	const userSettings = useSelector( getUserSettings ) ?? {};
+	const updateLanguage = ( event ) => {
```

Comment:
`/me/account` has a bunch of additional logic around `updateLanguage`:

https://github.com/Automattic/wp-calypso/blob/6e0933e052ac8d9aa6ec881208d31a3446a5df5a/client/me/account/main.jsx#L163-L203

What else of this is necessary to recreate for this particular context?

Reply:
```
if ( typeof empathyMode !== 'undefined' ) { 
 	this.updateUserSetting( 'i18n_empathy_mode', empathyMode ); 
}
```

I don't think `empathyMode` is totally relevant for our use case (#37106) but I've added support anyway: fd3ae3e725

```
if ( typeof useFallbackForIncompleteLanguages !== 'undefined' ) { 
	this.updateUserSetting( 
		'use_fallback_for_incomplete_languages', 
		useFallbackForIncompleteLanguages 
	); 
} 
```
`useFallbackForIncompleteLanguages` seems to be necessary and useful (#43571). Added support (and fixed bug in other changeset): 08cc3e4a4e

The other stuff seems related to the `/me/account` UI.

Reply:
For a dev helper for quickly switching languages, you probably want to copy the `QuickLanguageSwitcher` component as much as possible. It only calls the `setLocale` Redux action, setting the language for this particular session, overriding the user setting. And it doesn't persist the language, on reload you'll be back in the default language.